### DISCUSSION TITLE
Suggestions for including xit markup in "TODO" or "FIXME" code comments?

### DISCUSSION THREAD

**Discussion by shah** [jotaen/xit#30] [Other]
I love the idea of `xit` and was wondering if anyone is using it to formalize `TODO` and `FIXME` style markup in code comments? Many of us probably have code this like:

```js
// TODO: improve name of variable
const myVar = X();

// FIXME: implement this
function empty() {}
```

It would be great to include `xit` markup "inline" with the code and then be able to use parsers to pull that code out for other purposes. This might be a good feature to consider for `xit` specification. An obvious model might be something like the following:

```js
// TODO: [ ] This is an open item
const myVar = X();

// TODO: [x] This is a checked item      -- probably would never see this since finished items are usually removed
// TODO: [@] This is an ongoing item
// TODO: [~] This is an obsolete item
// FIXME: [ ] This is an open item
```

Alternatively, specifically include `(xit)` for parsers?

```js
// TODO(xit): [ ] This is an open item
const myVar = X();

// TODO(xit): [x] This is a checked item
// TODO(xit): [@] This is an ongoing item
// TODO(xit): [~] This is an obsolete item
// FIXME(xit): [x] This is a checked item
```

**Comment by jotaen** [comment]
Interesting idea, though I agree that I personally also tend to delete `TODO` comments once they are done.

To keep track of longer-running tasks in WIP projects, I sometimes put a `todo.xit` (formerly `todo.md`) in the project root.

**Reply by shah** [reply]
Agreed @jotaen these would be for long-running tasks and would facilitate the auto-generation of `todo.xit` style files.
